app/assets/javascripts/components/common/wiki_select.jsx

```diff
@@ -1,18 +1,141 @@
 import React from 'react';
-import createReactClass from 'create-react-class';
 import AsyncSelect from 'react-select/async';
 import PropTypes from 'prop-types';
 import { map } from 'lodash-es';
 import ArrayUtils from '../../utils/array_utils';
 import WIKI_OPTIONS from '../../utils/wiki_options';
-import { formatOption } from '../../utils/wiki_utils';
 
 /**
  *  A Wiki Selector Component that combines both language and project into a singular searchable
  *  component that works for both single-wiki and multi-wiki.
  */
-const WikiSelect = createReactClass({
-  propTypes: {
+const WikiSelect = ({
+  multi,
+  onChange,
+  styles,
+  wikis,
+  homeWiki,
+  readOnly,
+  label,
+  id,
+  options
+}) => {
+  const formatOption = (wiki) => {
+    return {
+      value: JSON.stringify(wiki),
+      label: url(wiki)
+    };
+  };
+
+  const url = (wiki) => {
+    const subdomain = wiki.language || 'www';
```

Comment:
This `language || 'www'` is duplicated here and is used in other places of the codebase. Maybe a good idea to refactor this into a function and use it everywhere

```
function getWikiSubdomain(wiki) {
  return wiki.language || 'www';
}
```